.circleci/compute_merge_commit.sh

```diff
@@ -21,30 +21,45 @@ set -e
 err=0
 
 GREEN() { echo -e "\n\033[0;32m$1\033[0m"; }
+CYAN() { echo -e "\033[0;36m$1\033[0m"; }
 
 # Ensure the CircleCI workspace directory exists.
 mkdir -p /tmp/workspace
 
 # Try to determine the PR number.
-curl -sS https://raw.githubusercontent.com/ampproject/amphtml/main/.circleci/get_pr_number.sh | bash
-if [[ -f "$BASH_ENV" ]]; then
+./.circleci/get_pr_number.sh
+if [[ -f "${BASH_ENV}" ]]; then
   source $BASH_ENV
 fi
 
 # If PR_NUMBER doesn't exist, there is nothing more to do.
-if [[ -z "$PR_NUMBER" ]]; then
+if [[ -z "${PR_NUMBER}" ]]; then
   exit 0
 fi
 
+echo "$(GREEN "Fetching all branches to update") $(CYAN ".git") $(GREEN "cache.")"
+git fetch
+
 # GitHub provides refs/pull/<PR_NUMBER>/merge, an up-to-date merge branch for
 # every PR branch that can be cleanly merged to the main branch. For more
 # details, see: https://discuss.circleci.com/t/show-test-results-for-prospective-merge-of-a-github-pr/1662
-MERGE_BRANCH="refs/pull/$PR_NUMBER/merge"
-echo $(GREEN "Computing merge SHA of $MERGE_BRANCH...")
-CIRCLE_MERGE_SHA="$(git ls-remote https://github.com/ampproject/amphtml.git "$MERGE_BRANCH" | awk '{print $1}')"
+MERGE_BRANCH="refs/pull/${PR_NUMBER}/merge"
+
+# Fetch the merge commit. This ensures that all CI stages use the same commit.
+echo "$(GREEN "Fetching the PR's merge branch") $(CYAN "${MERGE_BRANCH}")"
+(set -x && git pull --ff-only origin "${MERGE_BRANCH}") || err=$?
+
+# If a clean merge is not possible, do not proceed with the build. GitHub's UI
+# will show an error indicating there was a merge conflict.
+if [[ "$err" -ne "0" ]]; then
+  echo "$(RED "Detected a merge conflict between") $(CYAN "${CIRCLE_BRANCH}") $(RED "and the") $(CYAN "main") $(RED "branch.")"
+  echo "$(RED "Please rebase your PR branch.")"
+  exit $err
+fi
 
 # Store the merge commit info in the CircleCI workspace for use by followup
 # jobs.
+CIRCLE_MERGE_SHA="$(git rev-parse --verify HEAD)"
 echo "$CIRCLE_MERGE_SHA" > /tmp/workspace/.CIRCLECI_MERGE_COMMIT
-echo $(GREEN "Stored merge SHA $CIRCLE_MERGE_SHA in /tmp/workspace/.CIRCLECI_MERGE_COMMIT.")
+echo "$(GREEN "Stored merge SHA") $(CYAN "${CIRCLE_MERGE_SHA}") $(GREEN "in") $(CYAN "/tmp/workspace/.CIRCLECI_MERGE_COMMIT.")"
```

Comment:
These changes to the `echo` statements are causing incorrect logging (with unexpected newlines). Not sure if it's due to the extra leading and trailing quote, or due to the curly braces. Can we revert all the formatting changes in this PR for now?

![image](https://user-images.githubusercontent.com/26553114/127698351-a3fe5979-b2b5-436f-b069-30194d7496d9.png)

![image](https://user-images.githubusercontent.com/26553114/127698503-62c3c453-fd1b-47e7-808a-bd224d59c42a.png)

Reply:
For some reason `GREEN` and `RED` had `\n` in them. Fixed.